runner/logs.go

```diff
@@ -16,12 +16,16 @@ import (
 	"github.com/pkg/errors"
 )
 
-func downloadLogsForServer(server state.Server, globalCollectionOpts state.CollectionOpts, logger *util.Logger) (state.PersistedLogState, bool, error) {
+func downloadLogsForServer(server state.Server, globalCollectionOpts state.CollectionOpts, logger *util.Logger) (state.PersistedLogState, error) {
 	newLogState := server.LogPrevState
 
+	if server.LogPrevState.LogSnapshotDisabled {
+		return newLogState, errors.New(server.LogPrevState.LogSnapshotDisabledReason)
```

Comment:
I wonder if we shouldn't do `PrintWarning` here as well? (to match the stream-based log processing)

Reply:
Instead of or in addition to returning the error?

Reply:
Instead of, i.e. only output warning and not an error.